datafusion/physical-expr/src/window/lead_lag.rs

```diff
@@ -249,6 +252,9 @@ impl PartitionEvaluator for WindowShiftEvaluator {
                 // How many rows needed further than the current row to get necessary lead result
                 let offset: usize = self.non_null_offsets.iter().sum();
                 min(idx + offset + 1, n_rows)
+            } else if !self.ignore_nulls {
+                let offset = (-self.shift_offset) as usize;
+                min(idx + offset, n_rows)
```

Comment:
instead of `n_rows`, we now return `idx+offset`(can be smaller than `n_rows`, which will required less data for correct result) for `LEAD`